src/Changelog.js

```diff
@@ -28,107 +35,141 @@ export default class Changelog {
+
+          Object.keys(commitsByPackage).forEach(heading => {
+            markdown += "\n" + heading;
+
+            commitsByPackage[heading].forEach(commit => {
+              markdown += "\n  * ";
+
+              if (commit.number) {
+                const prUrl = this.remote.getBasePullRequestUrl() +
+                  commit.number;
+                markdown += "[#" + commit.number + "](" + prUrl + ") ";
+              }
+
+              if (commit.title.match(COMMIT_FIX_REGEX)) {
+                commit.title = commit.title.replace(
+                  COMMIT_FIX_REGEX,
+                  "Closes [#$3](" + this.remote.getBaseIssueUrl() + "$3)"
+                );
+              }
+
+              markdown += commit.title + "." + " ([@" + commit.user.login +
+                "](" +
+                commit.user.html_url +
+                "))";
+            });
+          });
+        });
 
-          if (commit.title.match(fixesRegex)) {
-            commit.title = commit.title.replace(fixesRegex, "Closes [#$3](" + this.remote.getBaseIssueUrl() + "$3)");
-          }
+      progressBar.terminate();
 
-          markdown += commit.title + "." + " ([@" + commit.user.login + "](" + commit.user.html_url + "))";
-        });
-      });
+      markdown += "\n\n#### Committers: " + committers.length + "\n";
+      markdown += committers.map(function(commiter) {
+        return "- " + commiter;
+      }).join("\n");
+      markdown += "\n\n\n";
     });
 
-    progressBar.terminate();
-
-    markdown += "\n\n#### Committers: " + committers.length + "\n";
-    markdown += committers.map(function(commiter) {
-      return "- " + commiter;
-    }).join("\n");
+    return markdown.substring(0, markdown.length - 3);
+  }
 
-    return markdown;
+  getListOfUniquePackages(sha) {
+    return Object.keys(
+      // turn into an array
+      execSync(
+        "git show -m --name-only --pretty='format:' --first-parent " + sha
+      )
+      .split("\n")
+      .reduce((acc, files) => {
+        if (files.indexOf("packages/") === 0) {
+          acc[files.slice(9).split("/", 1)[0]] = true;
+        }
+        return acc;
+      }, {})
+    );
   }
 
   getLastTag() {
     return execSync("git describe --abbrev=0 --tags");
   }
 
   getListOfCommits() {
-    var lastTag = this.getLastTag();
-    var commits = execSync("git log --oneline " + lastTag + "..").split("\n");
-    return commits;
+    // Determine the tags range to get the commits for. Custom from/to can be
+    // provided via command-line options.
+    // Default is "from last tag".
+    const tagFrom = this.tagFrom || this.getLastTag();
+    const tagTo = this.tagTo || "";
+    const tagsRange = tagFrom + ".." + tagTo;
+    const commits = execSync(
+      // Prints "<short-hash>;<ref-name>;<summary>;<date>"
+      // This format is used in `getCommitsInfo` for easily analize the commit.
+      'git log --oneline --pretty="%h;%D;%s;%cd" --date=short ' + tagsRange
```

Comment:
This is the most important part. Besides passing a custom range, the output is formatted with a specific pattern. This is necessary in order to analyse the commit.

```
<short-hash>;<ref-name>;<summary>;<date>
```